### ISSUE TITLE
[Bug] Note inclusion based on header anchor does not work with special chars in the header

### ISSUE THREAD

**Issue by dwrolvink** [obsidian-html/obsidian-html#303]
@dwrolvink Hello, it already worked perfectly for me 😀😀😀.
But I found some bugs in the links to blocks. For example using certain symbols will fail the build.
```md
%%% note1
# wifi 2.4 vs 5.0
---

# TCP/IP Model
---

# CI/CD && DevOps
---

%%% note2

Obsidian replaces these symbols with spaces
[[note1#wifi 2 4 vs 5 0]]  // fail
[[note1#TCP IP Model]] // fail
[[note1#CI CD DevOps]]  // fail
I had to change all those references for the build to work
```

_Originally posted by @GiancarloAparicio in https://github.com/obsidian-html/obsidian-html/issues/296#issuecomment-1197641425_

**Comment by dwrolvink** [comment]
Hi Giancarlo, 

I just made a new issue for this to track this separately, because it is a totally separate issue to #296 

The issue here is that Obsidian does not follow convention. For example:
header | python-markdown | obsidian | obsidian plus replace 
--------- | ------------------------ | ----------- | ----------------------------
`# wifi 2.4 vs 5.0` | `wifi-24-vs-50` | `wifi 2 4 vs 5 0`  | `wifi-2-4-vs-5-0` 

Note that obsidian replaces special characters with spaces, and spaces remain spaces. Where every other system that I know of removes the special chars and converts the spaces to dashes.

The issue is that after the fact I cannot tell whether a space represents a space or a special character. This is important because that will give a different header id when python-markdown processes the file.

I can probably rewrite the python-markdown behavior to generate different id's so that we use the same method of generating anchor links, but that is quite some work.

The good news is that for the note inclusion step, we fetch the entire page that is linked in order to be able to inject the included text, and we can just do some analysis to get the correct anchor anyways. I've added that just now, so that works for me now, despite all I said above.

One thing to note is that just linking to the same header will **not** work, as I don't want to fetch and parse every page that is linked to, obsidian-html is slow enough as it is.

I.e.:
``` bash
![[note1#wifi 2 4 vs 5 0]]   # note inclusion, will work
 [[note1#wifi 2 4 vs 5 0]]   # anchor link, will not work atm
```
